Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift

```diff
@@ -0,0 +1,135 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A control that displays an editable text interface for search inputs.
+///
+/// Use `inputState(_ state: State)` to change the state of the text field.
+public struct BPKSearchInputSummary: View {
+    public enum InputPrefix {
+        case text(String)
+        case icon(BPKIcon)
+    }
+    
+    struct Icon {
+        let icon: BPKIcon
+        let color: BPKColor
+    }
+    
+    @Binding private var text: String
+    @FocusState private var focused: Bool
+    private let placeholder: String
+    private let inputPrefix: InputPrefix
+    private var state: State = .default
+    
+    /// Creates a `BPKSearchInputSummary`.
+    ///
+    /// - Parameters:
+    ///   - placeholder: The placeholder text to display when the text field is empty.
+    ///   - inputPrefix: The prefix which would be displayed on the left of text input
+    ///   - text: The text to display in the text field.
+    public init(
+        placeholder: String = "",
+        inputPrefix: InputPrefix = .icon(.search),
+        _ text: Binding<String>
+    ) {
+        self.placeholder = placeholder
+        self.inputPrefix = inputPrefix
+        self._text = text
+    }
+    
+    public var body: some View {
+        HStack(spacing: .md) {
+            prefixView
+                .accessibilityHidden(true)
+            TextField(placeholder, text: $text)
+                .font(style: .bodyDefault)
+                .foregroundColor(state.textColor)
+                .disabled(state.isDisabled)
+                .focused($focused)
+            accessory
+        }
+        .padding(.md)
```

Comment:
Based on the design I think the padding is:
.padding(.left, .base)
.padding(.right, .md)
.padding(.vertical, .md)

Reply:
I thin it's actually `.padding(.horizontal, .base).padding(.vertical, BPKSpacing.md.value + BPKSpacing.sm.value)`  (we don't have a token for value 12 hehe) could you raise the pr so I can approve it myself?

Reply:
The padding is 16 (.base) from left [leading], and 12 (.md + .sm) for all other sides.
<img width="352" alt="Screenshot 2024-03-28 at 16 42 37" src="https://github.com/Skyscanner/backpack-ios/assets/159815080/a77bd504-77d6-416c-9b8e-e4de4378375c">
<img width="362" alt="Screenshot 2024-03-28 at 16 42 24" src="https://github.com/Skyscanner/backpack-ios/assets/159815080/ff1994a1-e77e-41d1-a92f-df76ae328394">

Please correct me if I read the Figma file incorrectly. we will make the PR for this change after that, no worries